scripts/release/prod/releases_page/update_releases_page

```diff
@@ -0,0 +1,183 @@
+            continue
+        didsuf = False
+        for suffix in (".asc", ".sig"):
+            if fname.endswith(suffix):
+                froot = fname[:-len(suffix)]
+                fd = files.get(froot)
+                if fd is None:
+                    fd = {}
+                    files[froot] = fd
+                fd[suffix] = x
+                didsuf = True
+                break
+        if didsuf:
+            continue
+        fd = files.get(fname)
+        if fd is None:
+            fd = {}
+            files[fname] = fd
+        fd["file"] = path
+        fd["Size"] = x["Size"]
+    return files
+
+def get_hashes_data(s3, rset):
+    text = ""
+    for x in rset:
+        path = x["Key"]
+        pre, fname = path.rsplit("/", 1)
+        if fname.endswith(".asc"):
+            continue
+        if fname.endswith(".sig"):
+            continue
+        if fname.startswith("hashes"):
+            ob = s3.get_object(Bucket=staging_bucket, Key=path)
+            text += ob["Body"].read().decode()
+    return text
+
+def read_hashes(fin):
+    by_fname = {}
+    for line in fin:
+        if not line:
+            continue
+        line = line.strip()
+        if not line:
+            continue
+        if line[0] == "#":
+            continue
+        hashstr, fname = line.split()
+        ob = by_fname.get(fname)
+        if not ob:
+            ob = {}
+            by_fname[fname] = ob
+        if len(hashstr) == 32:
+            ob["md5"] = hashstr
+        elif len(hashstr) == 64:
+            ob["sha256"] = hashstr
+        elif len(hashstr) == 128:
+            ob["sha512"] = hashstr
+    return by_fname
+
+def objects_by_fname(they):
+    out = {}
+    for x in they:
+        path = x["Key"]
+        if path.endswith("/"):
+            continue
+        parts = path.rsplit("/", 1)
+        fname = parts[-1]
+        out[fname] = x
+    return out
+
+def getContent(url):
+    with open(url, "r") as reader:
+        content = reader.read()
+
+    return content
+
+def build_page(channels):
+    html = getContent(html_tpl).replace("{styles}", getContent(styles_url))
+    html = html.replace("{stable}", "".join(channels["stable"])).replace("{beta}", "".join(channels["beta"]))
+    sys.stdout.write(html)
+
+def main():
+    s3 = boto3.client("s3")
+    channels = {}
+
+    for channel in ["stable", "beta"]:
+        staging_response = s3.list_objects_v2(Bucket=staging_bucket, Prefix="releases/" + channel + "/", MaxKeys=100)
+        release_sets = get_stage_release_set(staging_response)
+        releases_response = s3.list_objects_v2(Bucket=releases_bucket)
+        release_files = objects_by_fname(releases_response["Contents"])
+
+        table = []
+
+        for key in release_sets:
+            rset = release_sets[key]
+            hashftext = get_hashes_data(s3, rset)
+            fhashes = read_hashes(hashftext.splitlines())
+            files = release_set_files(rset)
+
+            def get_furl(fname, skey):
```

Comment:
either move this up to just after release_files is set, or move it up to file scope and add release_files as an argument

Reply:
Will change.